### ISSUE TITLE
How to specify the first day of week ?

### ISSUE THREAD

**Issue by jjavierdguezas** [arfedulov/semantic-ui-calendar-react#198]
**Describe the bug**
Is there a way to specify the first day of week?
just like `firstDayOfWeek` in [mdehoog/Semantic-UI-Calendar](https://github.com/mdehoog/Semantic-UI-Calendar#settings)

**Comment by jjavierdguezas** [comment]
@DovahBrownies maybe you can help me? thanks in advance

**Comment by DovahBrownies** [comment]
@jjavierdguezas hey! Sorry for the late reply... Sorry there doesn't seem to be a way to do that... I hope you've taken my earlier advice (well at least I think that I gave you the advice) of forking this project and editing it to your needs.

I've abandoned this project as the author isn't maintaining it and my employer doesn't need it anymore... What I _can_ do is at least give you some direction on how to do what you want to do...

<details>
<summary>Explanation of how the calendar is rendering the days/weeks</summary>
Let's take the **DayPicker** as an example. The DayPicker gets it's values by using the import `WEEKS_TO_DISPLAY` with the state `this.state.date`, and then builds/presents them using a function called `buildCalendarValues()`, [here](https://github.com/arfedulov/semantic-ui-calendar-react/blob/master/src/pickers/dayPicker/DayPicker.tsx#L78).

If you look at the **DayView**, you can see that it receives the value, [here](https://github.com/arfedulov/semantic-ui-calendar-react/blob/master/src/views/DayView.tsx#L62), and then immediately passes it to the CalendarBody.

In the **CalendarBody**, the values, now named the `data` prop, are presented and modified via the `buildRow()` function, [here](https://github.com/arfedulov/semantic-ui-calendar-react/blob/master/src/views/CalendarBody/Body.tsx#L51), and the table cells are finally returned to be rendered into the calendar you see.
</details>

TL;DR
I would look at the `buildCalendarValues()` function; More specifically - The function that it is using called `buildDays()`, shown [here](https://github.com/arfedulov/semantic-ui-calendar-react/blob/master/src/pickers/dayPicker/sharedFunctions.ts#L17).

Sorry, this is all I remember from the library, I could be wrong but I hope that this will help you get started if you decide to fork + modify.

**Comment by jjavierdguezas** [comment]
Thanks a lot @DovahBrownies ! And yes, I will take your advice and fork this project, I think it is the smartest thing to do

**Comment by ph-poppe** [comment]
@jjavierdguezas did you manage to modify the starting day in the end?

**Comment by jjavierdguezas** [comment]
> @jjavierdguezas did you manage to modify the starting day in the end?

@ph-poppe yes, we opted for an ugly "fix" for our particular problem (we needed to set Monday as the first day of the week): we just change the component localization to `en-gb` 😓 

It is ugly but easy to implement (as there is no need to modify the lib) and solves the problem with two lines of code

**Comment by ph-poppe** [comment]
@jjavierdguezas didn't get to work... so <DateInput localization="en-gb" .... /> would do the job? it stays on Sunday. Do you need to import other files from moment for this?

**Comment by jjavierdguezas** [comment]
> @jjavierdguezas didn't get to work... so <DateInput localization="en-gb" .... /> would do the job? it stays on Sunday. Do you need to import other files from moment for this?

sorry @ph-poppe , yes you need to add `import 'moment/locale/en-gb';` at the top of your file
